girder/models/model_base.py

```diff
@@ -45,6 +45,37 @@
 _modelSingletons = []
 
 
+def _permissionClauses(user=None, level=None, prefix=''):
+    """
+    Given a user and access level, return a list of clauses that can be used as
+    part of a Mongo find query or aggregate match step.
+
+    :param user: The user to check policies against.
+    :type user: dict or None
+    :param level: The access level.  Explicitly passing None skips doing
+        permissions checks.
+    :type level: AccessType
+    :param prefix: an optional string to prepend to the keys used in the
+        clauses.
+    :type prefix: str
+    :returns: A query dictionary with an '$or' entry which consists of a list
+        of match clauses, any one of which implies validation.
+    """
+    permissionClauses = [
+        {prefix + 'public': True},
```

Comment:
Since `public` only applies to `level=AccessType.READ`, should this be conditioned on the `level` parameter?